irods/client_configuration/__init__.py

```diff
@@ -161,18 +161,30 @@ def save(root = None, string='', file = ''):
 
 def _load_config_line(root, setting, value):
     arr = [_.strip() for _ in setting.split('.')]
+    loadexc = None
     # Compute the object referred to by the dotted name.
-    attr = ''
-    for i in filter(None,arr):
+    try:
+        attr = ''
+        for i in filter(None,arr):
+            if attr:
+                root = getattr(root,attr)
+            attr = i
+        # Assign into the current setting of the dotted name (effectively <root>.<attr>)
+        # using the loaded value.
         if attr:
-            root = getattr(root,attr)
-        attr = i
-    # Assign into the current setting of the dotted name (effectively <root>.<attr>)
-    # using the loaded value.
-    if attr:
-        return setattr(root, attr, ast.literal_eval(value))
+            return setattr(root, attr, ast.literal_eval(value))
+    except Exception as e:
+        loadexc = e
     error_message = 'Bad setting: root = {root!r}, setting = {setting!r}, value = {value!r}'.format(**locals())
-    raise RuntimeError (error_message)
+    if loadexc:
+        error_message += " [{loadexc!r}]".format(**locals())
+    if allow_config_load_errors:
+        #TODO: in the glorious Python3-only future, use raise X from Y to indicate causal chain
```

Comment:
link to the py3 issue https://github.com/irods/python-irodsclient/issues/480

Reply:
Look good?

Reply:
I guess I'm lazy in that regard and assume people will know what I mean ; ).  Yes, will be more specific

Reply:
So, I'm deleting the comment completely.  I've been on a 360-degree turn about this today, and decided finally the "raise from" syntax is not even appropriate here because we're not transforming one exception to another within an except clause (the Python version of a C++ "catch") , which was the main rationalization for the Python team in creating the  "raise from" mechanism in the first place... i.e. to avoid the confusing "During handling of the above exception, another exception occurred ...".  See [this](https://docs.python.org/3/tutorial/errors.html#exception-chaining)

I'm also reverting to the original logic now default behavior for PRC is not to raise the `RuntimeError` because it's fatal for the irods import, and that's not what a user typically wants.  Plus I've decided the logic is simpler and more correct if  opting-in for the RuntimeError to be generated should be a simple and predictable effect of the user's choice of value for the environment variable, `PYTHON_IRODSCLIENT_CONFIGURATION_LOAD_ERRORS_FATAL`

Reply:
We'll discuss at next standup but I  do believe things are now correct as they stand.

Reply:
Nice explanation. Agreed. Resolving.